### PR TITLE
Add new `auto-rotate-rke2-certs` setting

### PR DESCRIPTION
**Pull request by torchiaf**
### Summary

Related issue: https://github.com/harvester/harvester/issues/5176

#### PR Checklist
- Is this a multi-tenancy feature/bug?
    - [ ] Yes, the relevant RBAC changes are at:
- Do we need to backport changes to the [old Rancher UI](https://github.com/rancher/u), such as RKE1?
    - [ ] Yes, the relevant PR is at:
- Are backend engineers aware of UI changes?
    - [ ] Yes, the backend owner is:

Related issue: https://github.com/harvester/harvester/issues/5176

### Occurred changes and/or fixed issues

Adding a new setting

### Screenshot/Video

Settings page

image

Show setting details

image

Edit page

image

### CODE REVIEW

File: pkg/harvester/components/settings/auto-rotate-rke2-certs.vue [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        :labels="[t('generic.enabled'), t('generic.disabled')]"
+        @input="update"
+      />
+      <UnitInput
+        v-if="parseDefaultValue.enable"
+        v-model="parseDefaultValue.expiringInHours"
+        v-int-number
+        class="mb-20"
+        :min="1"
+        :max="8758"
```
Comment (a110605) [inline pkg/harvester/components/settings/auto-rotate-rke2-certs.vue:67]: [UnitInput component](https://github.com/a110605/harvester-dashboard/blob/1b7450ef2330ea05603606f26ed2754355474d31/shell/components/form/UnitInput.vue#L113) from shell doesn't have `max` prop only `min`.

User is able to input expireHour to 8759.
Backend error msg said the max number is 8759 not 8758.

Screenshot 2024-05-09 at 2 55 09 PM

Comment (FrankYang0529) [reply pkg/harvester/components/settings/auto-rotate-rke2-certs.vue:67]: The value should be `<= 8759`.

https://github.com/harvester/harvester/blob/5ab43f02144fb8c193005905945d60cc65a73133/pkg/webhook/resources/setting/validator.go#L935-L938

Comment (FrankYang0529) [reply pkg/harvester/components/settings/auto-rotate-rke2-certs.vue:67]: @torchiaf, sorry, I wrote the wrong value in description. Updated it.

Comment (a110605) [reply pkg/harvester/components/settings/auto-rotate-rke2-certs.vue:67]: So we can set `:max="8759"`

Comment (torchiaf) [reply pkg/harvester/components/settings/auto-rotate-rke2-certs.vue:67]: Good catch! Thank you.

**Review by a110605 — APPROVED** [review]
ＬＧＴＭ

**Comment by torchiaf** [comment]
@mergify backport release-harvester-v1.3

**Pull request merged**